src/main/java/com/epam/esm/util/validation/dao/GiftValidationDao.java

```diff
@@ -0,0 +1,58 @@
+package com.epam.esm.util.validation.dao;
+
+import com.epam.esm.persistence.dao.GiftCertificateDao;
+import com.epam.esm.persistence.util.search.GiftSearchFilter;
+import com.epam.esm.util.validation.BeforeGiftValidation;
+import com.epam.esm.web.exception.EntityBadInputException;
+import com.epam.esm.web.exception.EntityNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Component
+public class GiftValidationDao implements BeforeGiftValidation<GiftCertificateDao, Long> {
+
+    @Override
+    public void onBeforeDelete(Boolean isPresent, Long id) throws EntityNotFoundException{
+        if (!isPresent) {
+            throw new EntityNotFoundException("Requested gift not found (id = "+id+")");
+        }
+    }
+
+    @Override
+    public void onBeforeFindEntity(Optional<GiftCertificateDao> giftCertificateDao,
```

Comment:
Use Optional.orElseThrow instead of methods like this. The same applies to onBeforeUpdate method